### PR TITLE
openapi(dm): add error_msg in task status

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

Issue Number: ref #3583 

### What is changed and how it works?

as title

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes
 - Has exported variable/fields change

Side effects

 - Increased code complexity

Related changes

 - Need to update the documentation

### Release note 

```release-note
`None`.
```

### CODE REVIEW

File: dm/tests/openapi/client/openapi_task_check [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
+from pickle import TRUE
```
Comment (Ehco1996) [inline dm/tests/openapi/client/openapi_task_check:3]: remove

File: dm/tests/openapi/run.sh [hunk 2]
```diff
@@ -385,41 +385,58 @@ function test_task_templates() {
...
 
 	# check noshard task dump status success
 	openapi_task_check "check_noshard_task_dump_status_success" "$task_name" 0
 
-	# delete source success and clean data for other test
+	kill_dm_worker
+	check_port_offline $WORKER1_PORT 20
+	check_port_offline $WORKER2_PORT 20
+
+	openapi_task_check "get_task_status_success_but_worker_meet_error" "$task_name" 2
```
Comment (Ehco1996) [inline dm/tests/openapi/run.sh:426]: this line trigger the test

**Comment by Ehco1996** [comment]
could you please take another look for this pr ? thanks 🧡🧡🧡  

/cc @WizardXiao  @GMHDBJD

File: dm/dm/master/openapi.go [hunk 3]
```diff
@@ -940,14 +950,9 @@ func getOpenAPISubtaskStatusByTaskName(taskName string,
...
-		openapiSubTaskStatus := openapi.SubTaskStatus{
-			Name:                taskName,
-			SourceName:          sourceStatus.GetSource(),
-			WorkerName:          sourceStatus.GetWorker(),
-			Stage:               subTaskStatus.GetStage().String(),
-			Unit:                subTaskStatus.GetUnit().String(),
-			UnresolvedDdlLockId: &subTaskStatus.UnresolvedDDLLockID,
-		}
+		openapiSubTaskStatus.Stage = subTaskStatus.GetStage().String()
+		openapiSubTaskStatus.Unit = sourceStatus.GetSource()
```
Comment (WizardXiao) [inline dm/dm/master/openapi.go:954]:
```suggestion
		openapiSubTaskStatus.Unit = sourceStatus.GetUnit().String()
```

File: dm/dm/master/openapi.go [hunk 4]
```diff
@@ -929,6 +929,16 @@ func getOpenAPISubtaskStatusByTaskName(taskName string,
 			return nil, terror.ErrOpenAPICommonError.New("worker's query-status response is nil")
 		}
 		sourceStatus := workerStatus.SourceStatus
+		openapiSubTaskStatus := openapi.SubTaskStatus{
+			Name:       taskName,
+			SourceName: sourceStatus.GetSource(),
+			WorkerName: sourceStatus.GetWorker(),
+		}
+		if !workerStatus.Result {
```
Comment (WizardXiao) [inline dm/dm/master/openapi.go:937]: The logic here put construct `openapiSubTaskStatus` advance, and will lose other attributes like `Stage`, `Unit`. so just a question, does it meet expectations?

Comment (WizardXiao) [reply dm/dm/master/openapi.go:937]: oh, i see. `workerStatus.Result` is false when rpc to worker is fail. you test at integration test.

**Review by WizardXiao — COMMENTED** [review]
rest lgtm

**Comment by Ehco1996** [comment]
could you please take another look for this pr ? thanks 🧡🧡🧡

@WizardXiao @okJiang

**Review by WizardXiao — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**